server/container_create.go

```diff
@@ -488,6 +488,11 @@ func (s *Server) CreateContainer(ctx context.Context, req *pb.CreateContainerReq
 	}()
 
 	if _, err = s.ReserveContainerName(ctr.ID(), ctr.Name()); err != nil {
+		if cachedID := s.ResourceFromCache(ctr.Name()); cachedID != "" {
```

Comment:
We can guard this block against a specific typed error.

Reply:
what do you mean by that?